content/purgatorio/launch-and-mantain-the-system/automation-and-reproducibility.md

```diff
@@ -1,5 +1,242 @@
+- Reduces operation time and work handling time significantly.
+- Frees up workers to take on other roles.
+- It provides higher-level jobs in the development, deployment, maintenance, and running of the automated processes.
+
+This article explains very well why Automation is crucial for the Data Science world:
+
+- [4 Ways Automation Is Altering Data Science](https://towardsdatascience.com/4-ways-automation-is-altering-data-science-36a0d2755e4f)
+
+### Disadvantages of Automation
+
+Some might argue that Automation has not only advantages but also disadvantages, such as job losses (it takes less human to do the same things).
+
+This argument goes beyond the technical issue of how to improve the Data Science process, so it will not be covered in this guide. 
+
+However, this topic has valid counter-arguments, if you want to know more read here:
+
+- [Automating jobs is how society makes progress](https://qz.com/work/1212722/automating-jobs-is-how-society-makes-progress/)
+- [Will data science become automated?](https://www.reddit.com/r/datascience/comments/9mqghe/will_data_science_become_automated/)
+- [Can Data Science be Automated?](https://www.reddit.com/r/datascience/comments/bmk6cm/can_data_science_be_automated/)
+- [Do you think data science will get more automated and those without a PhD won't be competitive in the future?](https://www.reddit.com/r/datascience/comments/7f2m7i/do_you_think_data_science_will_get_more_automated/)
+
+So, we can state that Automation is crucial for the Data Science process to become useful, for a variety of reasons. But perhaps the most important reason, when it comes to automation in science, is the [**Reproducibility**](https://en.wikipedia.org/wiki/Reproducibility) issue. 
+
+**Data Science, precisely, is a "Science", in the sense that it must be demonstrable and reproducible!**
+
+::: warning
+Without **Reproducibility**, without the support of the _repeatability of experiments_, any result of Data Science is useless (or even harmful, because it can reinforce human bias).
+:::
+
+## Automation and Reproducibility for the Data Science Process
+
+The automation of Data Science steps is advisable at all levels, from the creation of the dataset to the deployment of trained ML models. 
+
+Most of the time, a configurable Python script is sufficient to automate most of the steps, which can for example be scheduled to be launched programmatically. 
+
+The following is a list of hints for some automation that can save you a lot of time (and therefore, money).
+
+You can automate:
+
+- **Data collection** 
+
+    Using scheduled scraping, database connections, or scheduled API, you can have always fresh data to train your models on. 
+
+- **Data quality control**
+
+    Using scripts that create statistical reports, for example on data distribution, the number of classes present, frequency, or other statistical values such as the most frequent data, average value, standard deviation. 
```

Comment:
Well said, @ericscheles says the same thing about how we can use descriptive statistics to analyse a dataframe/dataset to do the checks mentioned. See this resource for further insight into them https://gist.github.com/neomatrix369/9bd7196a46b4cdfe4d43f2065df08038

Reply:
There's also this Kolmogorov-Smirnov method which helps analyse two distributions and tell us if they have been drawn from the same continuous distribution